plugins/CoreAdminHome/Tasks.php

```diff
@@ -258,6 +262,48 @@ public function updateSpammerBlacklist()
         Option::set(ReferrerSpamFilter::OPTION_STORAGE_NAME, serialize($list));
     }
 
+    public function purgeOrphanedArchives()
+    {
+        $segmentHashesByIdSite = $this->getSegmentHashesByIdSite();
+        $archiveTables = ArchiveTableCreator::getTablesArchivesInstalled('numeric');
+
+        $datesPurged = array();
+        foreach ($archiveTables as $table) {
+            $date = ArchiveTableCreator::getDateFromTableName($table);
+            list($year, $month) = explode('_', $date);
+
+            $dateObj = Date::factory("$year-$month-15");
+
+            $this->archivePurger->purgeDeletedSiteArchives($dateObj);
+            $this->archivePurger->purgeDeletedSegmentArchives($dateObj, $segmentHashesByIdSite);
+
+            $datesPurged[$date] = true;
+        }
+    }
+
+    /**
+     * Get a list of all segment hashes that currently exist, indexed by idSite.
+     * @return array
+     */
+    public function getSegmentHashesByIdSite()
+    {
+        //Get a list of hashes of all segments that exist now
+        $sql = "SELECT DISTINCT definition, enable_only_idsite FROM " . Common::prefixTable('segment')
+            . " WHERE deleted = 0";
+        $rows = Db::fetchAll($sql);
+        $segmentHashes = array();
+        foreach ($rows as $row) {
+            try {
+                $segment = new Segment($row['definition'], array());
+                $segmentHashes[(int)$row['enable_only_idsite']][] = $segment->getHash();
```

Comment:
Some servers may trigger notices when not setting `$segmentHashes[(int)$row['enable_only_idsite']]` to an empty array first but not sure. Be good just to double check like 

```php
if (!isset($segmentHashes[(int)$row['enable_only_idsite']])) {
$segmentHashes[(int)$row['enable_only_idsite']] = array();
}
```

Reply:
OK have changed